File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    # 홀수이면 -1하고
    left = ""
    for i in range(1,len(food)):
        if food[i] % 2 == 1:
            food[i] -= 1
        left += str(i)*(food[i]//2)
    # 2로 나눈 몫 만큼 인덱스 문자열 추가            
    #print(left)
    # origin + 0 + reversed
    return left + '0' + left[::-1]

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public String solution(int[] food) {
        /*
        각 음식마다 2로 나눈 몫 만큼 문자열에 add한다 끝나면 0이랑 반대 붙이기
        */
        
        StringBuilder result = new StringBuilder();
        for(int i=1; i<food.length; i++){
            int cnt = food[i]/2;
            
            for(int j=0; j<cnt; j++){
                result.append(i);
            }
        }
        //log(result);
        result.append("0" + new StringBuilder(result).reverse());
        return result.toString();
    }
}